### PR TITLE
Build analysis

### PR DESCRIPTION
**Pull request by hassanfa**
This PR is far from what Maya asked me to do, but the complexity of mutliqc files will make it tough to work with Vogue. I've decided to strip down a lot of stuff from a valid multiqc.

`vogue load analysis -s 1 -a tests/fixtures/valid_multiqc.json -t QC` is to load sample-id `1`, but I get an error: 

```
runtimeerror: Working outside of application context.

This typically means that you attempted to use functionality that needed to interface with the current application object in some way. To solve this, set up an application context with app.app_context().  See the documentation for more information.
```

Now, how can I setup an app_context?

### CODE REVIEW

**Comment by mayabrandi** [comment]
Haha. Ive been debuging the same error all morning. With no success. Lets look at your case together

**Comment by hassanfa** [comment]
@mayabrandi Please have a look now, I made the changes we discussed. And thanks for the help! 😄

**Comment by mayabrandi** [comment]
ok! :)

File: vogue/commands/load/analysis.py [hunk 1]
```diff
@@ -2,17 +2,39 @@
...
 from vogue.build.analysis import build_analysis
+from vogue.tools.cli_utils import add_doc as doc
+from vogue.load.analysis import load_analysis
 import vogue.models.analysis as analysis_model
 
 LOG_LEVELS = ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL']
 LOG = logging.getLogger(__name__)
 
 
+def concat_dict_keys(my_dict: dict, key_name="", out_key_list=list()):
```
Comment (mayabrandi) [inline vogue/commands/load/analysis.py]: Maybe put this one in parse to?

File: vogue/load/analysis.py [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
-        raise SyntaxError("Sample {} does not exist".format(lims_id))
-    
-    ## TODO build the analysis object
-    analysis_obj = build_analysis(analysis, 'QC' )
-    ## TODO load the object with adapter
-    adapter.load_analysis(analysis_obj)
-    
-    return analysis_obj
+    if dry_run:
+        existing_sample = adapter.sample(lims_sample.id)
```
Comment (mayabrandi) [inline vogue/load/analysis.py]:
```suggestion
        existing_analysis = adapter.analysis(analysis['_id'])
```
Should be something like that? No?

File: vogue/load/analysis.py [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
+        if existing_sample:
+            LOG.info("The sample exists in the database")
+        else:
+            LOG.info("The sample does not exist in the database")
+
+        LOG.info("Sample information from analysis to add/update: \n %s", mongo_sample)
+        return
```
Comment (mayabrandi) [inline vogue/load/analysis.py]: Replace sample with analysis at several places above?

**Comment by mayabrandi** [comment]
Also, I think you need to update the analysis tests that are failing.

File: vogue/build/analysis.py [hunk 4]
```diff
@@ -1,4 +1,9 @@
 import logging
+import copy
+
+from genologics.entities import Sample
+from genologics.lims import Lims
+
```
[the remark this answers is not in the document]
Comment (hassanfa) [reply vogue/build/analysis.py]: ok, removed them! 👍

File: vogue/build/analysis.py [hunk 5]
```diff
@@ -12,29 +17,38 @@ def validate_conf(analysis_conf: dict):
     exist, disqualifies the file and returns False
     """
```
[the remark this answers is not in the document]
Comment (hassanfa) [reply vogue/build/analysis.py]: agreed, moved to parse.

File: vogue/commands/load/analysis.py (hunk 1 above)
Comment (hassanfa) [reply vogue/commands/load/analysis.py]: this might be more suitable to cli_utils. moved it there. 👍

File: vogue/load/analysis.py (hunk 2 above)
Comment (hassanfa) [reply vogue/load/analysis.py]: done!

File: vogue/load/analysis.py (hunk 3 above)
Comment (hassanfa) [reply vogue/load/analysis.py]: done!

**Pull request merged**